adobe/spectrum-css
# fix(textfield + stepper): allow is-disabled to supersede invalid and focus states

## Description

- [x] apply `.is-focused`, `:focus` and `is-invalid` states to `textfield` and dependent `stepper` components if `:not(.is-disabled)`. 

**Before:**

![Optional _thumb_12252805](https://github.com/user-attachments/assets/5a74570b-429a-4060-9c57-87f88c5b238d)
![Optional _thumb_12252804](https://github.com/user-attachments/assets/218be866-3e84-44d5-a283-a2e2cfb4f7a4)

**After:**

![Disabled stepper after change (via storybook)](https://github.com/user-attachments/assets/d085924e-85b0-47b1-888d-c8003aea86d3)
![Disabled textfield after change (via storybook)](https://github.com/user-attachments/assets/49e0ed06-ae7c-4321-afee-1ad7aa50d040)

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

1. Checkout `cdransf/textfield-is-disabled-conflicts` @marissahuysentruyt 
2. Run Storybook @marissahuysentruyt 
3. Access http://localhost:8080/?path=/story/components-stepper--default @marissahuysentruyt 
4. Verify the new test cases display appropriately for `Invalid + disabled + focused` and `Invalid + disabled + keyboard-focused` in the Chromatic test view.
5. Enable the `Disabled` state. Hover your mouse cursor over the component to verify that no border renders. @marissahuysentruyt 
6. Enable the `Focused` state. Verify that no outline renders. Repeat for `Keyboard focused.
7. Open http://localhost:8080/?path=/story/components-text-field--default and repeat the above state verification steps. @marissahuysentruyt 

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:


8. If components have been modified, VRTs have been run on this branch:


## Screenshots



## To-do list
